upb/def.c

```diff
@@ -2357,6 +2363,50 @@ static void create_service(
   }
 }
 
+static int popcount(uint64_t x) {
```

Comment:
the name popcount is not particularly legible (despite the asm precedent), nor does it convey assertion only.  I would prefer something more explicit here